src/PostgREST/DbStructure.hs

```diff
@@ -283,10 +283,8 @@ procsSqlQuery = [q|
     COALESCE(comp.relname, t.typname) AS name,
     p.proretset AS rettype_is_setof,
     (t.typtype = 'c'
-     -- Only pg pseudo type that is a row type is 'record'
-     or t.typtype = 'p' and t.typname = 'record'
-     -- if any INOUT or OUT arguments present, treat as composite
-     or COALESCE(proargmodes::text[] && '{b,o}', false)
+     -- if any TABLE, INOUT or OUT arguments present, treat as composite
+     or COALESCE(proargmodes::text[] && '{t,b,o}', false)
```

Comment:
Looks correct, since we cannot trust on the `t.typname = 'record'` being true, better rely on [proargmodes](https://www.postgresql.org/docs/13/catalog-pg-proc.html)(pg catalog considers the RETURNS part as arguments).